source/fe/fe_poly_tensor.cc

```diff
@@ -546,6 +654,7 @@ FE_PolyTensor<dim, spacedim>::fill_fe_values(
                   for (unsigned int k = 0; k < n_q_points; ++k)
                     for (unsigned int d = 0; d < dim; ++d)
                       output_data.shape_gradients[first + d][k] =
+                        fe_data.dof_sign_change[dof_index] * dof_sign *
```

Comment:
I think it would be easier to read if you initialized `dof_sign` further up with `fe_data.dof_sign_change[dof_index]` so that here you only have to multiply with one of these variables.

Reply:
done